vulture/core.py

```diff
@@ -218,7 +217,19 @@ def scan(self, code, filename=""):
             )
             self.found_dead_code_or_error = True
         else:
-            self.visit(node)
+            # When parsing type comments, visiting can throw SyntaxError.
+            try:
+                self.visit(node)
+            except SyntaxError as err:
+                self._handle_syntax_error(err, filename)
+
+    def _handle_syntax_error(self, err, filename):
```

Comment:
Let's make this a local function in the scan() method to encapsulate it better. Note that then the filename doesn't need to be passed.